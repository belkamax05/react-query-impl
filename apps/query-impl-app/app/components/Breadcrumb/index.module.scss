$breadcrumb-color-prev: hsla(34, 85%, 35%, 1);
$breadcrumb-color-curr: hsla(84, 85%, 35%, 1);

.root {
  & > * {
    padding: 8px;
    position: relative;
    background-color: $breadcrumb-color-prev;

    &:last-child {
      background-color: $breadcrumb-color-curr;
    }
    &:not(:first-child) {
      padding-left: 24px;
    }
    &::before {
      content: ' ';
      display: block;
      width: 0;
      height: 0;
      border-top: 20px solid transparent; /* Go big on the size, and let overflow hide */
      border-bottom: 20px solid transparent;
      border-left: 15px solid white;
      position: absolute;
      top: 50%;
      margin-top: -20px;
      margin-left: 1px;
      left: 100%;
      z-index: 1;
    }
    &::after {
      content: ' ';
      display: block;
      width: 0;
      height: 0;
      border-top: 20px solid transparent; /* Go big on the size, and let overflow hide */
      border-bottom: 20px solid transparent;
      border-left: 15px solid $breadcrumb-color-prev;
      position: absolute;
      top: 50%;
      margin-top: -20px;
      left: 100%;
      z-index: 2;
    }
    &:last-child {
      &::after {
        border-left-color: $breadcrumb-color-curr;
      }
      &::before {
        border-left-color: transparent;
      }
    }
  }
}
