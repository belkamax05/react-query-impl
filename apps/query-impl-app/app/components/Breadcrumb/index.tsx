'use client';
import _ from 'lodash';
import Link from 'next/link';
import { usePathname } from 'next/navigation';
import { useMemo } from 'react';
import classes from './index.module.scss';

const Breadcrumb = () => {
  const pathname = usePathname();
  const pathSplit = useMemo(
    () => pathname.split('/').filter((path) => !!path),
    [pathname]
  );

  return (
    <div className={classes.root}>
      {pathSplit.map((path, index) => {
        return (
          <Link key={path} href={`/${pathSplit.slice(0, index + 1).join('/')}`}>
            {_.capitalize(path)}
          </Link>
        );
      })}
    </div>
  );
};

export default Breadcrumb;
