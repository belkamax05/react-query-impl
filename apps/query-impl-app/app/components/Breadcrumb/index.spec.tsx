import { render } from '@testing-library/react';
import * as navigation from 'next/navigation';
import Breadcrumb from '.';

jest.mock('next/navigation');

const usePathnameSpy = jest.spyOn(navigation, 'usePathname');

describe('Breadcrumb', () => {
  beforeEach(() => {
    usePathnameSpy.mockReturnValue('/path/to/some/page');
  });

  it('should render the correct href for each breadcrumb link', () => {
    const { getByText } = render(<Breadcrumb />);
    expect(getByText('Path').getAttribute('href')).toBe('/path');
    expect(getByText('To').getAttribute('href')).toBe('/path/to');
    expect(getByText('Some').getAttribute('href')).toBe('/path/to/some');
    expect(getByText('Page').getAttribute('href')).toBe('/path/to/some/page');
  });
});
