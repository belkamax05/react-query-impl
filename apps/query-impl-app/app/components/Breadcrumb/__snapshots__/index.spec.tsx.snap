// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should render the correct href for each breadcrumb link 1`] = `
<body>
  <div>
    <div
      class="root"
    >
      <a
        href="/path"
      >
        Path
      </a>
      <a
        href="/path/to"
      >
        To
      </a>
      <a
        href="/path/to/some"
      >
        Some
      </a>
      <a
        href="/path/to/some/page"
      >
        Page
      </a>
    </div>
  </div>
</body>
`;
